storage/storage-interface/src/sharded_state_update_refs.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::{
+    state_store::{state_key::StateKey, state_value::StateValue, NUM_STATE_SHARDS},
+    write_set::WriteSet,
+};
+use arr_macro::arr;
+
+pub type StateUpdateRefWithOffset<'kv> = (usize, &'kv StateKey, Option<&'kv StateValue>);
+
+pub struct ShardedStateUpdateRefs<'kv> {
+    /// (idx, key, value) tuple per update, sharded
+    ///    Box<[]> to release over allocated memory during construction
```

Comment:
what does this comment mean?

Reply:
when `vec` converts to `box<[]>`, over allocated memory is freezed.